Mobile friendlier header, footer, and code blocks

jenweber opened the pull request:
Closes https://github.com/ember-learn/ember-api-docs/issues/317 and maybe https://github.com/ember-learn/ember-api-docs/issues/274

**Sidebar bug:** I brought in the media query breaks just a bit, to align the sidebar behavior with the container. This means it jumps into mobile arrangement sooner. 

**Responsive header/footer:** The main problem was that all the responsive CSS ported over from the main site wasn't being applied because that class was was not included in the html.

**Coderay block separating from line numbers:** just a global-ish css rule on tables that didn't make sense.

**Coderay block selection**: in the more global CSS was a `z-index: -1` applied to table layouts, which had an unexpected effect on code blocks. Setting the z index back to normal for coderay tables fixes the cursor problem. TBD whether code can be copied & pasted in mobile.

sivakumar-kailasam commented:
@jenweber let me know if it was because of page layout in which case this looks good to merge.

jenweber commented:
I'll change the breakpoints to match. Good catch.

jenweber commented:
@sivakumar-kailasam I have updated the header/footer breakpoints to match the rest of the website. It still needs another pair of eyes just to be sure. 

I have also replaced any stray breakpoints to be the variables in `_grid-settings.scss`. I changed the header nav to be relatively positioned. It was fixed and none of the rest of the site works that way. If that was intentional, we just need to ignore the final commit.

If it all looks right for you too, I think this is good to go.

sivakumar-kailasam commented:
Looks great now @jenweber. The header being fixed must've been a bug.
